### DISCUSSION TITLE
Adding sharepoint site but want to keep sync_dir

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#1773] [Q&A] [answered]
I have just installed the client, it's great, thank you! However, I have it syncing my main OneDrive files, but I also want to add a sharepoint site...and have a question:

Where the usage doc for Sharepoint says I should edit the config to "sync_dir = "~/SharePoint_My_Library_Name"

Won't this stop my client syncing everything that was previously kept at /OneDrive ? I want BOTH OneDrive main files AND my sharepoint site... is that not possible?

**Comment by abraunegg** [comment]
@johnsidoli 
> I want BOTH OneDrive main files AND my sharepoint site... is that not possible?

Yes, this is possible - but you have to run 2 client instances as this is *how* the OneDrive API operates.

Refer to: https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md for assistance.

**Comment by johnsidoli** [comment]
Hi thanks for this, I understand now!

However, I got the second sharepoint config working... but can't quite get my sharepoint library running as systemd... I have setup a normal onedrive sync as a system service (default config etc etc), and now I want to also make my new config for sharepoint "onedrive_sharepoint_accounts" (config name) into a systemd service (as user) as well.

I want to do this as my user (john) for systemd

I have browsed to /usr/lib/systemd/system and copied a new config onedrive_sharepoint_accounts@.service

I changed the config in it accordingly

Then I tried...

```
john@mint5:~$ systemctl --user enable onedrive_sharepoint_accounts
Failed to enable unit: Unit file onedrive_sharepoint_accounts.service does not exist.
```

and/or
```

john@mint5:~$ systemctl --user enable onedrive_sharepoint_accounts@john.service
Failed to enable unit: Unit file onedrive_sharepoint_accounts@john.service does not exist.
```

Do you know: What is going wrong?

EDIT: I can't actually tell if my original onedrive client is still running... it seems to be... but is there a command to check that?

Also... cheeky other question... do you take donations and/or do you think you will ever develop this to have a GUI? I think it's a great service!!!

**Comment by abraunegg** [comment]
@johnsidoli 
> Do you know: What is going wrong?

You need to refer to your Linux distribution support for help with systemd issues.

> EDIT: I can't actually tell if my original onedrive client is still running... it seems to be... but is there a command to check that?

You can use `systemctl status <service_name>` to see what the status of it is

> Also... cheeky other question... do you take donations and/or do you think you will ever develop this to have a GUI? I think it's a great service!!!

Ive tried to setup the github donations a while back - waiting on their review.

RE GUI .. not at this stage.

**Comment by johnsidoli** [comment]
Thanks! OneDrive appears to be working (changes get synced, and according to the config I have setup e.g. I put monitor interval right down to 5 and it clearly is using that config as I've tested and rapid changes get synced immediately), but my sharepoint config still isn't working as a service...

However, I don't understand this... systemct1 , according to your command above, gives the exact same output for both services........ apologies if this is beyond the scope of this discussion but could you take a guess at what is going on with the client here?

```
john@mint5:~$ systemctl status onedrive@john.service
● onedrive@john.service - OneDrive Free Client for john
     Loaded: loaded (/lib/systemd/system/onedrive@.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
       Docs: https://github.com/abraunegg/onedrive
```
That says disabled but the main OneDrive config I setup **is** definitely working...

Then the same output for my sharepoint config... also says disabled, but this config is **NOT** working (changes do **NOT** get synced, service is not working)

```
john@mint5:~$ systemctl status onedrive_sharepoint_accounts@john.service
● onedrive_sharepoint_accounts@john.service - OneDrive Free Client for john
     Loaded: loaded (/lib/systemd/system/onedrive_sharepoint_accounts@.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
       Docs: https://github.com/abraunegg/onedrive
```

One config working, the other isn't, but both have the same output - they say disabled on the output? But only the second one (sharepoint) isn't working...

**EDIT - ADDED POINT:**

I've been going through the instructions again as carefully as possible at Point 7 on advanced config:

https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#7-automatic-syncing-of-new-onedrive-configuration

...and there are several ambiguities that I can't get my head around:

1. It specifies a location for "RHEL / CentOS" and also "Others", but actually I am using Linux Mint and I have a onedrive service config file at all three locations... should I create a second one for the new config only at the top specified for 'Others' (for both or just for one?), or all three locations?

2. If only "Others: /usr/lib/systemd/user and /lib/systemd/system" ... do I put a new config file at BOTH locations or just one?

3. When it says to run the enable/start commands... do I do all or one or the other?

I think if these instructions were clearer then I'd probably have it setup by now, but because this is my first time using systemd config I am a bit confused at what the precise requirements are :(

Sorry if this is all my fault for doing stuff wrong.......!

**Comment by johnsidoli** [comment, marked as answer]
So, **I got it to work**. The above still stands (why does it say these services aren't working but the onedrive main service is working?). Also, the ambiguities I mention are still baffling me asto what you are meant to do exactly...

But here's what I did...

1. I followed the usage instructions and created my extra systemd file in /usr/lib/systemd/system as "onedrive-sharepoint-accounts@.service" and edited the ExecStart to ExecStart=/usr/bin/onedrive --monitor --confdir=/home/%i/.config/onedrive_sharepoint_accounts
User=%i
2. Then **instead** of this, which the usage instructions suggest:

```
systemctl --user enable onedrive-sharepoint-accounts
systemctl --user start onedrive-sharepoint-accounts
```

(Those commands generate the errors saying there's no file)

I remove "--user" and so I put these commands

```
systemctl enable onedrive-sharepoint-accounts
systemctl start onedrive-sharepoint-accounts
```

and it works! My service is up for **BOTH** the original onedrive config and also my sharepoint library called Accounts

Do you know why removing "--user" worked, or whether this will cause issues? I am on linuxmint... is this something that could cause issues later?

**Reply by abraunegg** [reply]
@johnsidoli 
> Do you know why removing "--user" worked, or whether this will cause issues? I am on linuxmint... is this something that could cause issues later?

It is to do with your Linux distribution.

Mint is based on Ubuntu which is based on Debian

Last time I checked, `--user` was only used by Debian / Ubuntu platforms - it may or may not be required for Mint .. which is why the documentation tries to show the differences because of all the shenanigans that Ubuntu add.

**Comment by johnsidoli** [comment]
Thanks!... so hopefully should work :smile:

**Comment by abraunegg** [comment]
@johnsidoli 
> Also... cheeky other question... do you take donations ...

GitHub has now approved sponsorship | donations - so this is now possible :)
